packages/ng-query/src/lib/query-client.ts

```diff
@@ -1,13 +1,55 @@
-import { inject, InjectionToken } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+import {
+  inject,
+  Injectable,
+  InjectionToken,
+  OnDestroy,
+  PLATFORM_ID,
+  Provider,
+} from '@angular/core';
 import { QueryClient as QueryCore } from '@tanstack/query-core';
 import { QUERY_CLIENT_OPTIONS } from './providers';
 
+const QueryClient = new InjectionToken<QueryCore>('QueryClient', {
+  providedIn: 'root',
+  factory() {
+    return new QueryCore(inject(QUERY_CLIENT_OPTIONS));
+  },
+});
+
+export const provideQueryClient = (queryClient: QueryCore): Provider => {
+  return {
+    provide: QueryClient,
+    useValue: queryClient,
+  };
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+class QueryClientHooks implements OnDestroy {
+  instance = inject(QueryClient);
+  isBrowser = isPlatformBrowser(inject(PLATFORM_ID));
+
+  constructor() {
+    if (this.isBrowser) {
+      this.instance.mount();
+    }
+  }
+  ngOnDestroy() {
+    if (this.isBrowser) {
+      this.instance.unmount();
+    }
+  }
+}
+
 export const QueryClientService = new InjectionToken<QueryCore>(
   'QueryClientService',
   {
     providedIn: 'root',
     factory() {
-      return new QueryCore(inject(QUERY_CLIENT_OPTIONS));
+      inject(QueryClientHooks);
```

Comment:
```ts
const hooks = inject(QueryClientHooks);
if( isPlatformBrowser(inject(PLATFORM_ID))) {
  hooks.init()
}
```

Reply:
```ts
@Injectable({
  providedIn: 'root',
})
class QueryClientHooks implements OnDestroy {
  instance = inject(QueryClient);
  initialized = false;

  init() {
    this.instance.mount();
    this.initialized = true;
  }

  ngOnDestroy() {
    if (this.initialized) {
      this.instance.unmount();
      this.initialized = false;
    }
  }
}
```
Would this be ok?

Reply:
why we need the check this.initialized

Reply:
Calling `unmount` on an unmounted instance could result in undefined behavior, just like calling `mount` without the `isPlatformBrowser` check could.